Assets/Scripts/PlayerController.cs

```diff
@@ -63,7 +91,7 @@ public void UpdateCamera()
     // Update is called once per frame
     private void Update()
     {
-        if (Input.GetAxis("Horizontal") != 0 || Input.GetAxis("Vertical") != 0)
+        if (Mathf.Abs(Input.GetAxis("Horizontal")) == 1 || Mathf.Abs(Input.GetAxis("Vertical")) == 1)
         {
             Move();
```

Comment:
Was this changed to remove minor movements which cause the spasms?

Reply:
its for removing the character slowly rotating by itself when you let go of buttons. makes it so that when next to a wall, the character doesn't sometimes push itself back by turning

Reply:
actually, I should mention that the new implementation is that in addition to stopping when running straight into a wall, when running into a wall at an angle, the player will now run alongside the wall.